crates/extensions/tedge_config_manager/src/lib.rs

```diff
@@ -88,9 +89,22 @@ impl ConfigManagerBuilder {
     }
 
     pub fn init(config: &ConfigManagerConfig) -> Result<(), FileError> {
+        if config.plugin_config_path.exists() {
+            return Ok(());
+        }
+
         // creating plugin config parent dir
         create_directory_with_defaults(&config.plugin_config_dir)?;
 
+        let legacy_plugin_config = config
+            .config_dir
+            .join("c8y")
+            .join("c8y-configuration-plugin.toml");
+        if legacy_plugin_config.exists() {
+            rename(legacy_plugin_config, &config.plugin_config_path)?;
```

Comment:
It would have been safer to use a non async version of `tedge_utils::file::move_file()`, but that's okay as it's so unlikely to have /etc/tedge/c8y and /etc/tedge/plugins on different filesystems.

Reply:
Yeah, that's exactly the reason why I went with this, to keep the diff small. I had started with that `move_file` API itself, but there were too many lines that I had to change (less risky ones, and mostly tests) to propagate that async call function call all the way up the call stack.